services/eventservice/src/data_manager.rs

```diff
@@ -0,0 +1,304 @@
+        manager
+    }
+
+    pub fn _merge(key: &[u8], old_v: Option<&[u8]>, new_v: &[u8]) -> Option<Vec<u8>> {
+        let mut old_vec = match old_v {
+            Some(v) => v.to_vec(),
+            None => Vec::new(),
+        };
+        old_vec.append(&mut new_v.to_vec());
+        Some(old_vec)
+    }
+
+    pub fn _compare_by_val(
+        key: Option<&String>,
+        a: &Event,
+        b: &Event,
+    ) -> Option<std::cmp::Ordering> {
+        match key {
+            Some(value) => match value.as_str() {
+                "title" => a.title.partial_cmp(&b.title),
+                "organizer" => a.organizer.partial_cmp(&b.organizer),
+                "date_created" => a.date_created.partial_cmp(&b.date_created),
+                "date_planning" => a.date_planning.partial_cmp(&b.date_planning),
+                &_ => a.date_planning.partial_cmp(&b.date_planning),
+            },
+            None => a.date_planning.partial_cmp(&b.date_planning),
+        }
+    }
+
+    pub fn _remove_id(data: &[u8], id: &[u8; 8]) -> Option<Vec<u8>> {
+        let mut data_vec = data.to_vec();
+        match data_vec.array_chunks::<8>().position(|&e| e.eq(id)) {
+            Some(index) => {
+                data_vec.drain(index..index + 8);
+                Some(data_vec)
+            }
+            None => None,
+        }
+    }
+
+    pub fn _events_to_vec(data: &Vec<&[u8]>) -> Vec<Event> {
+        let mut res = Vec::new();
+        data.iter().for_each(
+            |o| match Event::from_json(&String::from_utf8(o.to_vec()).unwrap()) {
+                Ok(parsed) => res.push(parsed),
+                Err(e) => eprintln!("{:?}", e),
+            },
+        );
+        res
+    }
+
+    pub fn _ids_to_vec(data: &[u8]) -> Vec<u64> {
+        data.array_chunks::<8>()
+            .map(|b| u64::from_be_bytes(*b))
+            .collect::<Vec<u64>>()
+    }
+
+    pub fn _reset_all(self) -> Result<bool, Box<dyn std::error::Error>> {
+        self.db.drop_tree(self.events_name)?;
+        self.db.drop_tree(self.organizers_name)?;
+        self.db.drop_tree(self.categories_name)?;
+        Ok(true)
+    }
+
+    pub fn generate_id(&self) -> Result<u64, Box<dyn std::error::Error>> {
+        match self.db.generate_id() {
+            Ok(id) => Ok(id),
+            Err(e) => Err(Box::new(e))
+        }
+    }
+
+    pub fn create_event(&self, new_event: &Event) -> Result<(), Box<dyn std::error::Error>> {
+        let id = new_event.id.to_be_bytes().clone();
+        let org_id = new_event.organizer.to_be_bytes().clone();
+        let cat = new_event.category.name.as_bytes();
+        let events = self.db.open_tree(&self.events_name)?;
+        events.insert(id.clone(), new_event.to_json().unwrap().as_bytes())?;
+        let organizers = self.db.open_tree(&self.organizers_name)?;
+        let categories = self.db.open_tree(&self.categories_name)?;
+        if organizers.contains_key(&org_id)? {
+            organizers.merge(&org_id, id)?;
```

Comment:
where do we actually store it?
Why do we use a bunch of vars instead of event object?

Reply:
We use array of bytes, cause this db works with them (see comment above). We store it locally